packages/flutter_tools/lib/src/context_runner.dart

```diff
@@ -88,11 +89,12 @@ Future<T> runInContext<T>(
     body: runnerWrapper,
     overrides: overrides,
     fallbacks: <Type, Generator>{
-      Analytics: () => Analytics(
-        tool: DashTool.flutterTool,
-        flutterChannel: globals.flutterVersion.channel,
-        flutterVersion: globals.flutterVersion.frameworkVersion,
-        dartVersion: globals.flutterVersion.dartSdkVersion,
+      Analytics: () => getAnalytics(
+        runningOnBot: runningOnBot,
+        flutterVersion: globals.flutterVersion,
+        suppressEnvFlag:
+            globals.platform.environment['FLUTTER_SUPPRESS_ANALYTICS'] ==
```

Comment:
Please review the style guide on formatting https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#formatting, dart format formatting is not allowed in flutter/flutter

Reply:
also, I would recommend just passing `globals.platform.environment` to `getAnalytics()`, and then allowing that function to pull out the env variable it needs. also, should we call .toLowerCase() on the env variable before comparing with `'true'`?

Reply:
> I would recommend just passing globals.platform.environment to getAnalytics()
That's a good idea! And agreed on making the env variable lowercase